From e7d391111ec2ab540639b5dc3a60ea33661e7a7c Mon Sep 17 00:00:00 2001
From: Nikita Popov <npopov@redhat.com>
Date: Fri, 4 Feb 2022 11:52:16 +0100
Subject: [PATCH] Work around gcc 12 crash while compiling
 intrinsics-library.cpp

This issue has already been fixed upstream by
https://gcc.gnu.org/git/gitweb.cgi?p=gcc.git;h=501c4ee9fad68716497db4d96b63b801cc57b4a1
but is not in the fedora rawhide build yet.

Temporarily work around the issue by explicitly specifying the
array size.
---
 flang/lib/Evaluate/intrinsics-library.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git -ruN a/flang-14.0.0rc1.src.orig/lib/Evaluate/intrinsics-library.cpp b/flang-14.0.0rc1.src/lib/Evaluate/intrinsics-library.cpp
--- a/flang-14.0.0rc1.src.orig/lib/Evaluate/intrinsics-library.cpp	2022-02-09 08:15:52.000000000 +0100
+++ b/flang-14.0.0rc1.src/lib/Evaluate/intrinsics-library.cpp	2022-02-28 10:36:55.652845443 +0100
@@ -207,7 +207,7 @@
 struct HostRuntimeLibrary<HostT, LibraryVersion::Libm> {
   using F = FuncPointer<HostT, HostT>;
   using F2 = FuncPointer<HostT, HostT, HostT>;
-  static constexpr HostRuntimeFunction table[]{
+  static constexpr HostRuntimeFunction table[22]{
       FolderFactory<F, F{std::acos}>::Create("acos"),
       FolderFactory<F, F{std::acosh}>::Create("acosh"),
       FolderFactory<F, F{std::asin}>::Create("asin"),
@@ -254,7 +254,7 @@
       const std::complex<HostT> &>;
   using F2B = FuncPointer<std::complex<HostT>, const std::complex<HostT> &,
       const HostT &>;
-  static constexpr HostRuntimeFunction table[]{
+  static constexpr HostRuntimeFunction table[18]{
       FolderFactory<F, F{std::acos}>::Create("acos"),
       FolderFactory<F, F{std::acosh}>::Create("acosh"),
       FolderFactory<F, F{std::asin}>::Create("asin"),
--
2.34.1

